main.go

```diff
@@ -297,19 +298,23 @@ func main() {
 	if err != nil {
 		l.WithError(err).Fatal("error getting current node")
 	}
-	routingMode, closeRoute, err := routing.LoadAndAttach(curNode, emulatorIP, serverPort)
+
+	routingMode, serverAddr, closeRoute, err := routing.LoadAndAttach(curNode, emulatorIP, serverPort)
 	if err != nil {
 		l.WithField("routing", routingMode).WithError(err).Fatal("error loading and attaching network route")
 	}
+
+	if attachNetworkRoutingAndExit {
```

Comment:
We should probably move this code much earlier and run it as soon as possible, in order to avoid running all the other code that is not required in the init container, like starting the watches and everything